src/ILLink.Tasks/CreateRuntimeRootDescriptorFile.cs

```diff
@@ -313,7 +317,7 @@ void OutputXml (string iLLinkTrimXmlFilePath, string outputFileName)
 					featureAssemblyNode.Attributes.Append (featureDefault);
 
 					foreach (string typeName in featureSwitchMembers[fsMembers].Keys) {
-						AddXmlTypeNode (doc, featureAssemblyNode, typeName, featureSwitchMembers[fsMembers]);
+						AddXmlTypeNode (doc, featureAssemblyNode, typeName, featureSwitchMembers[fsMembers][typeName]);
```

Comment:
This can be rewritten like this to simplify it even more:
```C#
foreach (var type in featureSwitchMembers[fsMembers]) {
    AddXmlTypeNode (doc, featureAssemblyNode, type.Key, type.Value);
}
```